pkg/imgutil/imgutil.go

```diff
@@ -102,6 +103,52 @@ func GetExistingImage(ctx context.Context, client *containerd.Client, snapshotte
 	return res, nil
 }
 
+func EnsureAllContent(ctx context.Context, client *containerd.Client, rawRef string, target ocispec.Descriptor, platform string, options types.GlobalCommandOptions) error {
+	named, err := distributionref.ParseDockerRef(rawRef)
+	if err != nil {
+		return err
+	}
+	refDomain := distributionref.Domain(named)
+	var platformComparer platforms.MatchComparer
+	if platform != "" {
+		parsed, err := platforms.Parse(platform)
+		if err != nil {
+			return err
+		}
+		platformComparer = platformutil.NewMatchComparerFromOCISpecPlatformSlice([]ocispec.Platform{parsed})
+	} else {
+		platformComparer = platforms.Default()
+	}
+
+	_, _, _, missing, err := images.Check(ctx, client.ContentStore(), target, platformComparer)
+	if err != nil {
+		return err
+	}
+
+	if len(missing) > 0 {
+		// Get a resolver
+		var dOpts []dockerconfigresolver.Opt
+		if options.InsecureRegistry {
+			log.G(ctx).Warnf("skipping verifying HTTPS certs for %q", refDomain)
+			dOpts = append(dOpts, dockerconfigresolver.WithSkipVerifyCerts(true))
+		}
+		dOpts = append(dOpts, dockerconfigresolver.WithHostsDirs(options.HostsDir))
+		resolver, err := dockerconfigresolver.New(ctx, refDomain, dOpts...)
+		if err != nil {
+			return err
+		}
+
+		opts := []containerd.RemoteOpt{
+			containerd.WithResolver(resolver),
+			containerd.WithPlatformMatcher(platformComparer),
+		}
+		_, err = client.Fetch(ctx, rawRef, opts...)
```

Comment:
Is there any output when fetching layers?

Reply:
Do we want this 'processing' outputs?

Reply:
There probably is output.
I don't know if it is desirable to show it or not... if we do show it, it will change behavior on the output of commands and diverge from docker (especially since it is supposed to go to stdout - #3395) - if we do not show it, it might appear to the user as if it is hanging...

wdyt?

Reply:
Last push does show the output.
Still not sure if we want it or not :-), but at least that gives us options.

Reply:
no idea :-).